File: app/src/main/java/dev/ragnarok/fenrir/model/CommunityDetails.java
package dev.ragnarok.fenrir.model;

import java.util.List;


public class CommunityDetails {

    private int allWallCount;

    private int ownerWallCount;

    private int postponedWallCount;

    private int suggestedWallCount;

    private boolean canMessage;

    private boolean isFavorite;

    private boolean isSubscribed;

    private int topicsCount;

    private int docsCount;

    private int photosCount;

    private int audiosCount;

    private int videosCount;

    private int articlesCount;

    private int productsCount;

    private int chatsCount;

    private String status;

    private Audio statusAudio;

    private Cover cover;

    private String description;

    public Cover getCover() {
        return cover;
    }

    public CommunityDetails setCover(Cover cover) {
        this.cover = cover;
        return this;
    }

    public int getChatsCount() {
        return chatsCount;
    }

    public CommunityDetails setChatsCount(int chatsCount) {
        this.chatsCount = chatsCount;
        return this;
    }

    public boolean isSetFavorite() {
        return isFavorite;
    }

    public CommunityDetails setFavorite(boolean isFavorite) {
        this.isFavorite = isFavorite;
        return this;
    }

    public boolean isSetSubscribed() {
        return isSubscribed;
    }

    public CommunityDetails setSubscribed(boolean isSubscribed) {
        this.isSubscribed = isSubscribed;
        return this;
    }

    public int getAllWallCount() {
        return allWallCount;
    }

    public CommunityDetails setAllWallCount(int allWallCount) {
        this.allWallCount = allWallCount;
        return this;
    }

    public int getOwnerWallCount() {
        return ownerWallCount;
    }

    public CommunityDetails setOwnerWallCount(int ownerWallCount) {
        this.ownerWallCount = ownerWallCount;
        return this;
    }

    public int getPostponedWallCount() {
        return postponedWallCount;
    }

    public CommunityDetails setPostponedWallCount(int postponedWallCount) {
        this.postponedWallCount = postponedWallCount;
        return this;
    }

    public int getSuggestedWallCount() {
        return suggestedWallCount;
    }

    public CommunityDetails setSuggestedWallCount(int suggestedWallCount) {
        this.suggestedWallCount = suggestedWallCount;
        return this;
    }

    public boolean isCanMessage() {
        return canMessage;
    }

    public CommunityDetails setCanMessage(boolean canMessage) {
        this.canMessage = canMessage;
        return this;
    }

    public Audio getStatusAudio() {
        return statusAudio;
    }

    public CommunityDetails setStatusAudio(Audio statusAudio) {
        this.statusAudio = statusAudio;
        return this;
    }

    public String getStatus() {
        return status;
    }

    public CommunityDetails setStatus(String status) {
        this.status = status;
        return this;
    }

    public int getTopicsCount() {
        return topicsCount;
    }

    public CommunityDetails setTopicsCount(int topicsCount) {
        this.topicsCount = topicsCount;
        return this;
    }

    public int getDocsCount() {
        return docsCount;
    }

    public CommunityDetails setDocsCount(int docsCount) {
        this.docsCount = docsCount;
        return this;
    }

    public int getPhotosCount() {
        return photosCount;
    }

    public CommunityDetails setPhotosCount(int photosCount) {
        this.photosCount = photosCount;
        return this;
    }

    public int getArticlesCount() {
        return articlesCount;
    }

    public CommunityDetails setArticlesCount(int articlesCount) {
        this.articlesCount = articlesCount;
        return this;
    }

    public int getProductsCount() {
        return productsCount;
    }

    public CommunityDetails setProductsCount(int productsCount) {
        this.productsCount = productsCount;
        return this;
    }

    public int getAudiosCount() {
        return audiosCount;
    }

    public CommunityDetails setAudiosCount(int audiosCount) {
        this.audiosCount = audiosCount;
        return this;
    }

    public int getVideosCount() {
        return videosCount;
    }

    public CommunityDetails setVideosCount(int videosCount) {
        this.videosCount = videosCount;
        return this;
    }

    public String getDescription() {
        return description;
    }

    public CommunityDetails setDescription(String description) {
        this.description = description;
        return this;
    }

    public static final class Cover {

        private boolean enabled;

        private List<CoverImage> images;

        public List<CoverImage> getImages() {
            return images;
        }

        public Cover setImages(List<CoverImage> images) {
            this.images = images;
            return this;
        }

        public boolean isEnabled() {
            return enabled;
        }

        public Cover setEnabled(boolean enabled) {
            this.enabled = enabled;
            return this;
        }
    }

    public static final class CoverImage {

        private final String url;

        private final int height;

        private final int width;

        public CoverImage(String url, int height, int width) {
            this.url = url;
            this.height = height;
            this.width = width;
        }

        public int getHeight() {
            return height;
        }

        public int getWidth() {
            return width;
        }

        public String getUrl() {
            return url;
        }
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/model/CommunityDetails.kt
package dev.ragnarok.fenrir.model

class CommunityDetails {
    private var allWallCount = 0
    private var ownerWallCount = 0
    private var postponedWallCount = 0
    private var suggestedWallCount = 0
    private var canMessage = false
    private var isFavorite = false
    private var isSubscribed = false
    private var topicsCount = 0
    private var docsCount = 0
    private var photosCount = 0
    private var audiosCount = 0
    private var videosCount = 0
    private var articlesCount = 0
    private var productsCount = 0
    private var chatsCount = 0
    private var productServicesCount = 0
    private var narrativesCount = 0
    private var status: String? = null
    private var statusAudio: Audio? = null
    private var cover: Cover? = null
    private var description: String? = null

    fun setProductServicesCount(productServicesCount: Int): CommunityDetails {
        this.productServicesCount = productServicesCount
        return this
    }

    fun setNarrativesCount(narrativesCount: Int): CommunityDetails {
        this.narrativesCount = narrativesCount
        return this
    }

    fun getProductServicesCount(): Int {
        return productServicesCount
    }

    fun getNarrativesCount(): Int {
        return narrativesCount
    }

    fun getCover(): Cover? {
        return cover
    }

    fun setCover(cover: Cover?): CommunityDetails {
        this.cover = cover
        return this
    }

    fun getChatsCount(): Int {
        return chatsCount
    }

    fun setChatsCount(chatsCount: Int): CommunityDetails {
        this.chatsCount = chatsCount
        return this
    }

    fun isSetFavorite(): Boolean {
        return isFavorite
    }

    fun setFavorite(isFavorite: Boolean): CommunityDetails {
        this.isFavorite = isFavorite
        return this
    }

    fun isSetSubscribed(): Boolean {
        return isSubscribed
    }

    fun setSubscribed(isSubscribed: Boolean): CommunityDetails {
        this.isSubscribed = isSubscribed
        return this
    }

    fun getAllWallCount(): Int {
        return allWallCount
    }

    fun setAllWallCount(allWallCount: Int): CommunityDetails {
        this.allWallCount = allWallCount
        return this
    }

    fun getOwnerWallCount(): Int {
        return ownerWallCount
    }

    fun setOwnerWallCount(ownerWallCount: Int): CommunityDetails {
        this.ownerWallCount = ownerWallCount
        return this
    }

    fun getPostponedWallCount(): Int {
        return postponedWallCount
    }

    fun setPostponedWallCount(postponedWallCount: Int): CommunityDetails {
        this.postponedWallCount = postponedWallCount
        return this
    }

    fun getSuggestedWallCount(): Int {
        return suggestedWallCount
    }

    fun setSuggestedWallCount(suggestedWallCount: Int): CommunityDetails {
        this.suggestedWallCount = suggestedWallCount
        return this
    }

    fun isCanMessage(): Boolean {
        return canMessage
    }

    fun setCanMessage(canMessage: Boolean): CommunityDetails {
        this.canMessage = canMessage
        return this
    }

    fun getStatusAudio(): Audio? {
        return statusAudio
    }

    fun setStatusAudio(statusAudio: Audio?): CommunityDetails {
        this.statusAudio = statusAudio
        return this
    }

    fun getStatus(): String? {
        return status
    }

    fun setStatus(status: String?): CommunityDetails {
        this.status = status
        return this
    }

    fun getTopicsCount(): Int {
        return topicsCount
    }

    fun setTopicsCount(topicsCount: Int): CommunityDetails {
        this.topicsCount = topicsCount
        return this
    }

    fun getDocsCount(): Int {
        return docsCount
    }

    fun setDocsCount(docsCount: Int): CommunityDetails {
        this.docsCount = docsCount
        return this
    }

    fun getPhotosCount(): Int {
        return photosCount
    }

    fun setPhotosCount(photosCount: Int): CommunityDetails {
        this.photosCount = photosCount
        return this
    }

    fun getArticlesCount(): Int {
        return articlesCount
    }

    fun setArticlesCount(articlesCount: Int): CommunityDetails {
        this.articlesCount = articlesCount
        return this
    }

    fun getProductsCount(): Int {
        return productsCount
    }

    fun setProductsCount(productsCount: Int): CommunityDetails {
        this.productsCount = productsCount
        return this
    }

    fun getAudiosCount(): Int {
        return audiosCount
    }

    fun setAudiosCount(audiosCount: Int): CommunityDetails {
        this.audiosCount = audiosCount
        return this
    }

    fun getVideosCount(): Int {
        return videosCount
    }

    fun setVideosCount(videosCount: Int): CommunityDetails {
        this.videosCount = videosCount
        return this
    }

    fun getDescription(): String? {
        return description
    }

    fun setDescription(description: String?): CommunityDetails {
        this.description = description
        return this
    }

    class Cover {
        private var enabled = false
        private var images: ArrayList<CoverImage>? = null
        fun getImages(): ArrayList<CoverImage>? {
            return images
        }

        fun setImages(images: ArrayList<CoverImage>?): Cover {
            this.images = images
            return this
        }

        fun isEnabled(): Boolean {
            return enabled
        }

        fun setEnabled(enabled: Boolean): Cover {
            this.enabled = enabled
            return this
        }
    }

    class CoverImage(private val url: String?, private val height: Int, private val width: Int) {
        fun getHeight(): Int {
            return height
        }

        fun getWidth(): Int {
            return width
        }

        fun getUrl(): String? {
            return url
        }
    }
}